jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/DelayedHandler.java

```diff
@@ -103,44 +99,33 @@ protected Request.Processor delayed(Request request, Request.Processor processor
             if (!request.getConnectionMetaData().getHttpConfiguration().isDelayDispatchUntilContent())
                 return processor;
 
-            Charset charset = FutureFormFields.getFormEncodedCharset(request);
+            Charset charset = FormFields.getFormEncodedCharset(request);
             if (charset == null)
                 return processor;
 
-            return new UntilFormFieldsProcessor(request, processor, charset);
+            return new UntilFormFieldsProcessor(request, processor);
         }
     }
 
     private static class UntilFormFieldsProcessor implements Request.Processor, BiConsumer<Fields, Throwable>
     {
         private final Request _request;
         private final Request.Processor _processor;
-        private final Charset _charset;
         private Response _response;
         private Callback _callback;
 
-        public UntilFormFieldsProcessor(Request request, Request.Processor processor, Charset charset)
+        public UntilFormFieldsProcessor(Request request, Request.Processor processor)
```

Comment:
Removed `Charset charset`?
How is the HTML special form field `_charset_` handled now?

Reply:
I'm not aware of `_charset_` for `application/x-www-form- urlencoded`. Do you have a reference?
I only saw it for multipart.

As for the removal of the charset, it is a pre-check to avoid delaying if the content-type is not `application/x-www-form- urlencoded`. If it is, the charset will be computed by `FormFields`.
I think it's wrong to be able to create an instance of `UntilFormFieldsProcessor` with a `Charset`, because it may not match the one present in the request headers.